### PR TITLE
ADD fermat's-Little-theorem in mathematics for multiplicative modulo …

### LINKED ISSUE #144: [Feature Request] <ADD Fermat's Little Theorem In Mathematics For Modulo Multiplicative Inverse>
### Describe the solution you'd like
About theorem: Fermat's Little Theorem is useful for finding the modular inverse of a number when the modulus is a prime number. The theorem states:

a^(M-1) ≡ 1 (mod M)
Where:

a is any integer,
M is a prime number,
a^(M-1) means "a raised to the power (M-1)."
From this, we can derive the multiplicative inverse of a modulo M as:

a^(-1) ≡ a^(M-2) (mod M)
Thus, to compute the modular inverse of a under modulo M, you can calculate:

a^(-1) = a^(M-2) (mod M)
This can be efficiently done using modular exponentiation, typically by the method of exponentiation by squaring.

### Describe alternatives you've considered
-

### Additional context
Time Complexity O(logn)

---

**Would you like to work on this feature?**  
- Yes

### PR DESCRIPTION
**Pull request by kRajoria121**
…inverse

### Description
ADD fermat's-Little-theorem in mathematics for multiplicative modulo inverse

Fixes # 144

### Type of change

-  New feature (non-breaking change which adds functionality)

### Checklist:

-  My code follows the style guidelines of this project
-  I have performed a self-review of my code
- I have commented my code, particularly in hard-to-understand areas
-  I have added tests that prove my fix is effective or that my feature works
-  New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/mathematics-algorithms/fermat's-Little-Theorem.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
+---
+id: fermat's-Little-theorem
```
Comment (ajay-dhangar) [inline docs/mathematics-algorithms/fermat's-Little-Theorem.md]: replace `id: fermat's-Little-theorem` with `id: fermat-little-theorem`

File: docs/mathematics-algorithms/fermat's-Little-Theorem.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+title: Fermat's Little Theorem
+sidebar_label: Fermat's Little Theorem MMI
```
Comment (ajay-dhangar) [inline docs/mathematics-algorithms/fermat's-Little-Theorem.md]: replace `sidebar_label: Fermat's Little Theorem MMI` with `sidebar_label: "Fermat's Little Theorem MMI"`

File: docs/mathematics-algorithms/fermat's-Little-Theorem.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+sidebar_position: 5
+description: A comprehensive guide to calculating the modular multiplicative inverse using Fermat's Little Theorem.
```
Comment (ajay-dhangar) [inline docs/mathematics-algorithms/fermat's-Little-Theorem.md]: replace `description: A comprehensive guide to calculating the modular multiplicative inverse using Fermat's Little Theorem.` with `description: "A comprehensive guide to calculating the modular multiplicative inverse using Fermat's Little Theorem."`

File: docs/mathematics-algorithms/fermat's-Little-Theorem.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+## Modular Multiplicative Inverse Using Fermat's Little Theorem
+
+**Definition:**
+
+The modular multiplicative inverse of an integer `a` modulo `m` is an integer `b` such that `a * b ≡ 1 (mod m)`. In other words, `b` is the inverse of `a` in the ring of integers modulo `m`.
+
+**Fermat's Little Theorem for Modular Inverses:**
+
+Fermat's Little Theorem provides a direct way to calculate the modular multiplicative inverse of `a` modulo `p` when `p` is a prime number. It states that:
+a^(p-2) ≡ a^(-1) (mod p)
```
Comment (ajay-dhangar) [inline docs/mathematics-algorithms/fermat's-Little-Theorem.md]: replace `a^(p-2) ≡ a^(-1)` with `$a^{(p-2)} ≡ a^{(-1)}$`

File: docs/mathematics-algorithms/fermat's-Little-Theorem.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+This means that raising `a` to the power of `p-2` modulo `p` gives you the modular multiplicative inverse of `a`.
+<!-- 
+### Remember
+(A/B)%mod != ((A%mod)/(B%mod))%mod why ? do experiments :) -->
```
Comment (ajay-dhangar) [inline docs/mathematics-algorithms/fermat's-Little-Theorem.md]: remove comment

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

File: docs/mathematics-algorithms/fermat's-Little-Theorem.md (hunk 5 above)
Comment (kRajoria121) [reply docs/mathematics-algorithms/fermat's-Little-Theorem.md]: Check both which is better ALL DONE or Changes Done

File: .vscode/c_cpp_properties.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (ajay-dhangar) [inline .vscode/c_cpp_properties.json]: delete `.vscode/c_cpp_properties.json` file

File: docs/mathematics-algorithms/fermat-little-theorem.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+        System.out.println("The modular multiplicative inverse of " + a + " modulo " + p + " is " + inverse);
+    }
+}
+```
+Explanation of the Code:
+
+modular_exponentiation: This function efficiently calculates the modular exponentiation of base^exponent mod mod using the repeated squaring algorithm.
+modular_inverse: This function calculates the modular multiplicative inverse of a modulo p using Fermat's Little Theorem.
+It first checks if p is a prime number greater than 1.
+If so, it calculates a^(p-2) mod p using the modular_exponentiation function.
```
Comment (ajay-dhangar) [inline docs/mathematics-algorithms/fermat-little-theorem.md]: replace `a^(p-2)` with `$a^{(p-2)}$`

File: docs/mathematics-algorithms/fermat-little-theorem.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
+---
+id: fermat-little-theorem
+title: Fermat's Little Theorem
```
Comment (ajay-dhangar) [inline docs/mathematics-algorithms/fermat-little-theorem.md]: also, replace `title: Fermat's Little Theorem` with `title: "Fermat's Little Theorem"`

File: .vscode/settings.json [hunk 9]
```diff
@@ -0,0 +1,59 @@
+{
```
Comment (ajay-dhangar) [inline .vscode/settings.json]: delete `.vscode/settings.json` file

File: .vscode/launch.json [hunk 10]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (ajay-dhangar) [inline .vscode/launch.json]: delete `.vscode/launch.json` file

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by kRajoria121** [comment]
thanks for guiding and corporating sir

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**